manifests/conf.pp

```diff
@@ -27,12 +32,22 @@
   String $setting,
   Scalar $value,
   String $confdir,
-  String $section = 'main',
+  String $section = $setting ? { 'environment' => 'agent', default => 'main' },
   Enum['present', 'absent'] $ensure = 'present',
 ) {
-
   $l_name = "${module_name}_${name}"
 
+  if $setting == 'environment' {
+    if !defined(Ini_setting['remove environment from main']) {
+      ini_setting { 'remove environment from main':
+        ensure  => absent,
+        path    => "${confdir}/puppet.conf",
+        section => 'main',
+        setting => 'environment'
+      }
+    }
+  }
+
```

Comment:
There should be a way to force-set `environment` within `main` if the user wishes to do so.  Perhaps add a `$pupmod::master::allow_environment_in_main` param that defaults to `false` ?

I'll try that...

Reply:
Pushed a commit for review.